### PR TITLE
 Added code of daily Coding Exercise-20

### PR DESCRIPTION
**Pull request by anushreeSonak**

### CODE REVIEW

File: First/dailyExercises/EvenLength.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+                // Print the word
+                System.out.println(word);
+    }
+
+    public static void main(String[] args) {
+        String data = "Hello World I am Anushree S Sonak";
+        printWords(data);
+    }
+}
+//OUTPUT:"C:\Program Files\Java\jdk-17\bin\java.exe" "-javaagent:C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2023.1\lib\idea_rt.jar=58799:C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2023.1\bin" -Dfile.encoding=UTF-8 -classpath "C:\Users\anushrees\Desktop\New folder\DemoRepo\First\out\production\First1" firstPackage.First.dailyExercises.EvenLength
```
Comment (AmrutaDhage20) [inline First/dailyExercises/EvenLength.java]: Add Readme file for Question and output.

**Review by AmrutaDhage20 — CHANGES_REQUESTED** [review]

File: First/dailyExercises/EvenLength.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package firstPackage.First.dailyExercises;
+
+public class EvenLength {
+    public static void printWords(String data) {
+        // Splits Str
+        for (String word : data.split(" "))
+            // if length is even
+            if (word.length() % 2 == 0)
+                // Print the word
```
Comment (AmrutaDhage20) [inline First/dailyExercises/EvenLength.java:9]: Explain me the code.

**Review by AmrutaDhage20 — CHANGES_REQUESTED** [review]

File: First/dailyExercises/EvenLength.java (hunk 2 above)
Comment (anushreeSonak) [reply First/dailyExercises/EvenLength.java:9]: Done !! Explained the code to Amruta.

File: First/dailyExercises/EvenLength.java (hunk 1 above)
Comment (anushreeSonak) [reply First/dailyExercises/EvenLength.java]: Done !! Added Read Me file

File: First/dailyExercises/EvenLength.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package firstPackage.First.dailyExercises;
+
+public class EvenLength {
+    public static void printWords(String data) {
+        // Splits Str
+        for (String word : data.split(" "))
+            // if length is even
+            if (word.length() % 2 == 0)
```
Comment (amitsahu8) [inline First/dailyExercises/EvenLength.java:8]: not a medium complexity code

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

**Review by AmrutaDhage20 — APPROVED** [review]

File: First/dailyExercises/StringRotationCheck.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package firstPackage.First.dailyExercises;
+
```
Comment (AmrutaDhage20) [inline First/dailyExercises/StringRotationCheck.java]: Give proper package/subpackage name.

File: First/dailyExercises/StringRotationCheck.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+public class StringRotationCheck {
+    public static boolean areRotations(String data, String mockData) {
```
Comment (AmrutaDhage20) [inline First/dailyExercises/StringRotationCheck.java]: Add space between method and class.

File: First/dailyExercises/StringRotationCheck.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        // Check if both strings have the same length
+        if (data.length() != mockData.length()) {
+            return false;
+        }
+        // Concatenate str1 with itself
+        String concatenated = data + data;
+        // Check if str2 is a substring of the concatenated string
```
Comment (AmrutaDhage20) [inline First/dailyExercises/StringRotationCheck.java]: Explain the code.

File: First/dailyExercises/StringRotationCheck.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    public static void main(String[] args) {
+        String data = "abcd";
+        String mockData = "cdab";
+        if (areRotations(data, mockData)) {
+            System.out.println("The strings are rotations of each other.");
+        } else {
+            System.out.println("The strings are not rotations of each other.");
+        }
+    }
+}
```
Comment (AmrutaDhage20) [inline First/dailyExercises/StringRotationCheck.java]: Add output in READme file.

**Review by AmrutaDhage20 — CHANGES_REQUESTED** [review]

File: First/dailyExercises/StringRotationCheck.java (hunk 6 above)
Comment (anushreeSonak) [reply First/dailyExercises/StringRotationCheck.java]: Done !!

File: First/dailyExercises/StringRotationCheck.java (hunk 5 above)
Comment (anushreeSonak) [reply First/dailyExercises/StringRotationCheck.java]: Done!

File: First/dailyExercises/StringRotationCheck.java (hunk 7 above)
Comment (anushreeSonak) [reply First/dailyExercises/StringRotationCheck.java]: Done !

**Review by AmrutaDhage20 — APPROVED** [review]

File: First/dailyExercises/StringRotationCheck.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+public class StringRotationCheck {
+
+    public static boolean areRotations(String data, String mockData) {
+        // Check if both strings have the same length
+        if (data.length() != mockData.length()) {
+            return false;
+        }
+        // Concatenate str1 with itself
+        String concatenated = data + data;
```
Comment (amitsahu8) [inline First/dailyExercises/StringRotationCheck.java]: explain the question

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

File: First/dailyExercises/StringRotationCheck.java (hunk 8 above)
Comment (anushreeSonak) [reply First/dailyExercises/StringRotationCheck.java]: The problem statement of the above program is to determine whether two strings are rotations of each other.

**Review by prashantkawa — APPROVED** [review]

**Pull request merged**